bitswap/server/internal/decision/engine.go

```diff
@@ -294,6 +298,24 @@ func WithSetSendDontHave(send bool) Option {
 	}
 }
 
+// WithNotifyNewBlocks sets or not whether to notify peers when receiving a block.
+// By default, it is true. This can be useful if you want the server to only
+// reply if it has a block at the moment it receives a message, and not later.
+func WithNotifyNewBlocks(notify bool) Option {
+	return func(e *Engine) {
+		e.notifyNewBlocks = notify
+	}
+}
+
+// WithNoPeerLedger disables the usage of a peer ledger to track want lists. This
+// means that this engine will not keep track of the CIDs that peers wanted in
+// the past.
+func WithNoPeerLedger() Option {
```

Comment:
@lidel that was actually my original idea. I didn't do it at the time because it'll need some refactoring around the types. But I can take a better look tody.